src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -569,6 +569,17 @@ public BoundNode VisitStatement(BoundNode node)
             return VisitSideEffect(node);
         }
 
+        public ImmutableArray<BoundStatement> VisitSideEffects(ImmutableArray<BoundStatement> statements)
+        {
+            var result = ArrayBuilder<BoundStatement>.GetInstance(statements.Length);
+            foreach (BoundStatement statement in statements)
+            {
+                result.Add((BoundStatement)VisitSideEffect(statement));
```

Comment:
>result.Add((BoundStatement)VisitSideEffect(statement)); [](http://example.com/codeflow?start=16&length=55)

Usually, in cases like this, we track if any changes were made by ```VisitSideEffect``` and allocate the builder only then.